mlir/lib/Dialect/MemRef/IR/MemRefOps.cpp

```diff
@@ -2356,12 +2356,11 @@ LogicalResult ExpandShapeOp::verify() {
   // Verify if provided output shapes are in agreement with output type.
   DenseI64ArrayAttr staticOutputShapes = getStaticOutputShapeAttr();
   ArrayRef<int64_t> resShape = getResult().getType().getShape();
-  unsigned staticShapeNum = 0;
-
-  for (auto [pos, shape] : llvm::enumerate(resShape))
-    if (!ShapedType::isDynamic(shape) &&
-        shape != staticOutputShapes[staticShapeNum++])
+  for (auto [pos, shape] : llvm::enumerate(resShape)) {
+    if (!ShapedType::isDynamic(shape) && shape != staticOutputShapes[pos]) {
       emitOpError("invalid output shape provided at pos ") << pos;
+    }
```

Comment:
The above needs to be a `return emitOpError`.  Also is it possible to just add a test for this? The original PR didnt fail on anything so the issue here was missed.

Reply:
The format as IIUC the `output_shape` is a mix of static and dynamic. So you should be able to try
```
 %r = memref.expand_shape %0 [[0, 1], [2]] output_shape [%sz0, %sz1, 16]
        : memref<?x32xf32> into memref<?x?x32xf32>
```
to check (hopefully)

Reply:
this occurred to me but I thought "if *that* is broken then this is basically unusable". maybe it is... will try.

Reply:
Yup, reproduces.

And (I don't know how I didn't notice earlier), but testing this also revealed an existing test case that was incorrect. fixed.